src/components/previewCard/index.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { FC } from "react"
+
+import styles from "./index.module.css"
+import UploadImagesIcon from "../icons/uploadImage"
+import ArrowRightTeal from "../icons/arrowRight"
+import { FixedCard } from "../card/fixed"
+
+interface Props {
+  name: string
+  location: {
+    address?: string
+    zipCode: string
+    city?: string
+  }
+  title?: string
+  previewLabel: string
+  image?: string
+  logo?: string
+  link?: string
+  editMode?: boolean
+  handleOnClick?: () => void
+  handleOnError?: (e) => void
+  handleOnClickText?: () => void
+}
+
+const PreviewCard: FC<Props> = ({
+  name,
+  location,
+  title,
+  previewLabel,
+  image,
+  logo,
+  link,
+  editMode,
+  handleOnClick,
+  handleOnError,
+  handleOnClickText,
+}) => {
+  return (
+    <FixedCard
+      renderImage={() => (
+        <div className="bg-grey-1 flex justify-center items-center text-grey-2 md:h-cardImage w-12/12 md:w-card">
+          {!image && editMode ? (
+            <UploadImagesIcon width="72px" height="72px" />
+          ) : (
+            <img
+              src={image}
+              className={`${styles.promotionImage} w-12/12 object-cover`}
+              alt={title}
+              onClick={() => {
+                handleOnClick && handleOnClick()
+              }}
+              onError={(e) => {
+                handleOnError && handleOnError(e)
+              }}
+            />
+          )}
+        </div>
+      )}
+    >
+      <div
+        className="p-20 w-12/12 md:w-card max-h-dropdown"
+        onClick={() => {
+          handleOnClickText && handleOnClickText()
+        }}
+      >
+        {title ? (
+          <p className="text-lg">{title}</p>
+        ) : (
+          <>
+            <div className="bg-grey-1 h-20 w-12/12" />
+            <div className="bg-grey-1 h-20 w-12/12 my-10" />
+            <div className="bg-grey-1 h-20 w-6/12" />
```

Comment:
what are these 3 empty divs doing?

Reply:
It was used in DH, when there's no title it displays grey lines. From what I understood it was like a placeholder. I can add an if statement to show it only in DH ?

Reply:
I see. generally, the component should be agnostic to where it's used and it should not know any details of the project where it's placed. so, a conditional to check which project it is used in would be a breach of this principle

if the content in a shared component is supposed to be flexible, [render props](https://reactjs.org/docs/render-props.html) are often a good choice. the DH-specific markup would be passed in DH, whereas the listings-specific one would be passed in listings. this way, the component itself does not need to know about the details of what exactly the content looks like, but it reserves a specific place to include it

Reply:
I get it ! I will drop `editMode` and try with render props

Reply:
we discussed to keep it as is
we didn't really find a viable solution to take everything apart nicely